opennlp-tools/src/main/java/opennlp/tools/parser/Parse.java

```diff
@@ -518,6 +518,7 @@ public void add(Parse daughter, HeadRules rules) {
     this.head = rules.getHead(getChildren(),type);
     if (head == null) {
       System.err.println(parts);
+      throw new IllegalArgumentException("Rule returned null head!");
```

Comment:
ok, let me look more at this, otherwise I suggest we pull it out here. In my experience you should never do changes during code cleanup that are not 100% safe to do.